### DISCUSSION TITLE
create3 Python docking and undocking

### DISCUSSION THREAD

**Discussion by jdbrubaker** [iRobotEducation/create3_docs#270] [Questions]
Hi, I am able to dock and undock my robot with the cli commands, but the Python examples given in the Python playground do not work.  Please tell me what the proper Python functions or commands are for docking and undocking.  thanks ---john.

**Comment by shamlian** [comment]
Hi there -- I'd like to try to help, but I have some questions. What version of firmware is your robot running? What "cli commands" are you referring to? And when you say the Python playground does not work, do other parts of the Python playground work (are you able to connect and run other commands, like basic movement commands)? I suspect you may just be running older firmware that does not support docking over BLE. Per the [protocol spec](https://github.com/iRobotEducation/root-robot-ble-protocol#device-19---docking-sensors) you need to be running protocol version 1.5 which requires firmware version [G.3 or greater](https://github.com/iRobotEducation/root-robot-ble-protocol#ble-protocol-version).

If other commands in the Python playground do not work:
- Are you using a browser that supports Web Bluetooth? As far as I know, the only browsers that do are Chromium-based.
- Have you clicked the "connect" button and connected successfully to the robot (it will play an ascending tone once you do)?
